### PR TITLE
Dev 5739 exclude unrevealed submission periods from award rollups

### PR DESCRIPTION
**Pull request by BrianZito**
**Description:**
Adds a join to the award rollup and idv rollup to prevent totals from being aggregated if they belong to a submission that is not past the reveal date.

**Technical details:**
This exends to agency counts, and the federal account count as well. These were beyond the scope of dev 5739, but still things we wanted to do.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated (N/A)
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [ ] Jira Ticket [DEV-5739](https://federal-spending-transparency.atlassian.net/browse/DEV-5739):
    - [x] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

**Comment by BrianZito** [comment]
After talking with @collinwr, we decided to not merge this PR until https://github.com/fedspendingtransparency/usaspending-api/pull/2793 is merged, then update the logic matching submissions to submission windows.

File: usaspending_api/awards/tests/data/award_test_data.py [hunk 1]
```diff
@@ -17,150 +18,190 @@
 }
 OBLIGATED_AMOUNT_BY_AWARD_ID = {0: 0.0, 1: 110011.0, 2: 430043.0, 3: 960096.0, 4: 1700170.0}
 
+DATE_IN_THE_PAST = "1776-07-04"
+DATE_IN_THE_FUTURE = "2553-04-01"
```
Comment (tony-sappe) [inline usaspending_api/awards/tests/data/award_test_data.py:22]: Oddly the French Embassy in the US has nothing scheduled for that date https://franceintheus.org/spip.php?page=MF-jour&date=2553-04-01

Comment (BrianZito) [reply usaspending_api/awards/tests/data/award_test_data.py:22]: Good catch. I will notify the french at once.

**Review by tony-sappe — APPROVED** [review]
Wow, lots of additions to the tests.

File: usaspending_api/awards/v2/views/funding_rollup.py [hunk 2]
```diff
@@ -16,10 +16,7 @@
     with gather_financial_accounts_by_awards as (
         WITH closed_submissions AS (
```
Comment (tony-sappe) [inline usaspending_api/awards/v2/views/funding_rollup.py:17]: Is `closed_submission` necessary? Look at the SQL changes in #2793

Comment (BrianZito) [reply usaspending_api/awards/v2/views/funding_rollup.py]: It is not. The SQL has been updated, and there might be some other queries throughout the codebase we can update in the tech debt sprint.

File: usaspending_api/awards/v2/views/funding_rollup.py [hunk 3]
```diff
@@ -20,7 +20,9 @@
                 faba.treasury_account_id
         from    awards a
                 inner join financial_accounts_by_awards faba on faba.award_id = a.id
-        where   {award_id_column} = {award_id}
+                INNER JOIN submission_attributes sa ON faba.submission_id = sa.submission_id
+                INNER JOIN dabs_submission_window_schedule dabs ON sa.submission_window_id = dabs.id
+        where   {award_id_column} = {award_id} and dabs.submission_reveal_date <= now()
```
Comment (tony-sappe) [inline usaspending_api/awards/v2/views/funding_rollup.py:25]: The other places changed in #2793 using this new FK places `submission_reveal_date <= now()` into the join clause. Did you discover performance issues doing that here?

Comment (BrianZito) [reply usaspending_api/awards/v2/views/funding_rollup.py:25]: I did not. I will test performance of that strategy in order to maintain consistency.

**Review by tony-sappe — APPROVED** [review]

**Pull request merged**